couchdb/test.couchdb-cluster.yml

```diff
@@ -0,0 +1,85 @@
+version: "3.3"
+networks:
+  cht-net-test:
+    driver: bridge
+
+services:
+  couchdb.1:
+    build:
+      context: .
+      target: cht-couchdb
+    environment:
+      - "COUCHDB_USER=${COUCHDB_USER:-admin}"
+      - "COUCHDB_PASSWORD=${COUCHDB_PASSWORD:-password}"
+      - "COUCHDB_SECRET=${COUCHDB_SECRET:-6c1953b6-e64d-4b0c-9268-2528396f2f58}"
+      - "COUCHDB_UUID=${COUCHDB_UUID:-5c265815-b9e3-47f1-ba8d-c1d50495eeb2}"
+      - "SVC_NAME=couchdb.1"
+      - "CLUSTER_PEER_IPS=couchdb.2,couchdb.3"
+    restart: always
+    networks:
+      cht-net-test:
+
+  couchdb.2:
+    build:
+      context: .
+      target: cht-couchdb
+    environment:
+      - "COUCHDB_USER=${COUCHDB_USER:-admin}"
+      - "COUCHDB_PASSWORD=${COUCHDB_PASSWORD:-password}"
+      - "COUCHDB_SECRET=${COUCHDB_SECRET:-6c1953b6-e64d-4b0c-9268-2528396f2f58}"
+      - "COUCHDB_UUID=${COUCHDB_UUID:-5c265815-b9e3-47f1-ba8d-c1d50495eeb2}"
+      - "SVC_NAME=couchdb.2"
+    restart: always
+    networks:
+      cht-net-test:
+
+  couchdb.3:
+    build:
+      context: .
+      target: cht-couchdb
+    environment:
+      - "COUCHDB_USER=${COUCHDB_USER:-admin}"
+      - "COUCHDB_PASSWORD=${COUCHDB_PASSWORD:-password}"
+      - "COUCHDB_SECRET=${COUCHDB_SECRET:-6c1953b6-e64d-4b0c-9268-2528396f2f58}"
+      - "COUCHDB_UUID=${COUCHDB_UUID:-5c265815-b9e3-47f1-ba8d-c1d50495eeb2}"
+      - "SVC_NAME=couchdb.3"
+    restart: always
+    networks:
+      cht-net-test:
+
+  haproxy:
+    image: medicmobile/cht-haproxy:archv3-test6
+    hostname: haproxy
+    environment:
+      - "HAPROXY_IP=${HAPROXY_IP:-haproxy}"
+      - "COUCHDB_USER=${COUCHDB_USER:-admin}"
+      - "COUCHDB_PASSWORD=${COUCHDB_PASSWORD:-password}"
+      - "COUCHDB1_SERVER=${COUCHDB1_SERVER:-couchdb.1}"
+      - "COUCHDB2_SERVER=${COUCHDB2_SERVER:-couchdb.2}"
+      - "COUCHDB3_SERVER=${COUCHDB3_SERVER:-couchdb.3}"
+      - "HAPROXY_PORT=${HAPROXY_PORT:-5984}"
+    ports:
+      - "${HAPROXY_PORT:-5984}:5984"
```

Comment:
Ah yeah! it is being hardcoded, and that will be a quick change I'll include in the HAProxy PR

Reply:
I take the above statement back, I hadn't updated github, but the cht-haproxy image `archv3-test6` contains configurable HAPROXY_PORT. I created the cluster, ran docker-compose down, exported a new HAPROXY_PORT, and restarted the template with success. 

```
docker exec -it couchdb_couchdb.2_1 /bin/bash
root@2c1904713865:/# curl http://admin:password@haproxy:5985/_membership
{"all_nodes":["couchdb@couchdb.1","couchdb@couchdb.2","couchdb@couchdb.3"],"cluster_nodes":["couchdb@couchdb.1","couchdb@couchdb.2","couchdb@couchdb.3"]}
```

Reply:
Can you run tests with a different `HAPROXY_PORT`? 
I tried accessing haproxy from `localhost` and failed with a different port, succeeded with the hardcoded one.

Reply:
I've found it! It only effects tests, and i ran the above on the prod template. 

I'm going to remove HAPROXY from the test suite, since its not being used but the environment variables overlap and will interfere `check_if_couchdb_cluster_is_ready()` in `tests.sh`

Reply:
I'v removed HAProxy from test suite for clustered couchdb

Reply:
I also couldn't reach the cluster from localhost with the prod docker-compose, with a custom haproxy port.

Reply:
In order for that scenario to work, you need to `export HAPROXY_IP=0.0.0.0` then launch the prod template. This will bind haproxy to localhost inside the container, which makes the export port statement in the prod template work correctly. 

We won't be binding to 0.0.0.0 (or exposing Haproxy), so that statement will be removed. It was leftover from debugging.

Reply:
I've removed the exposed port for HAProxy. If you wanted to expose that to debug/test, you would want to add this, along with exporting HAPROXY_IP:
```
    ports:
      - "${HAPROXY_PORT:-5984}:${HAPROXY_PORT:-5984}"
```
at the bottom of the haproxy declaration.

Reply:
But with the default haproxy port I was able to access haproxy through localhost, and was able to point my local cht to use that as the database (without editing the whole docker-compose file and building images).